radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/CandidateForkVotesPostProcessor.java

```diff
@@ -0,0 +1,192 @@
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.statecomputer;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.hash.HashCode;
+import com.radixdlt.application.validators.state.ValidatorSystemMetadata;
+import com.radixdlt.atom.SubstateTypeId;
+import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.consensus.bft.BFTValidatorSet;
+import com.radixdlt.constraintmachine.REProcessedTxn;
+import com.radixdlt.constraintmachine.RawSubstateBytes;
+import com.radixdlt.constraintmachine.SubstateDeserialization;
+import com.radixdlt.constraintmachine.SubstateIndex;
+import com.radixdlt.engine.PostProcessor;
+import com.radixdlt.engine.PostProcessorException;
+import com.radixdlt.serialization.DeserializeException;
+import com.radixdlt.statecomputer.forks.CandidateForkVote;
+import com.radixdlt.statecomputer.forks.ForkVotingResult;
+import com.radixdlt.store.EngineStore;
+import com.radixdlt.utils.Bytes;
+import com.radixdlt.utils.Pair;
+import com.radixdlt.utils.UInt256;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/** Adds forksVotingResults at epoch boundary to the result metadata. */
+public final class CandidateForkVotesPostProcessor implements PostProcessor<LedgerAndBFTProof> {
+  /**
+   * Specifies a minimum stake that needs to vote for a fork to be added to the resulting stored
+   * map. Acts as a storage size optimization so that we don't store insignificant entries.
+   */
+  private static final short VOTES_THRESHOLD_TO_STORE_RESULT = 5000; /* 50.00% */
```

Comment:
Maybe you could add a comment explaining why 5000 means 50%?

Reply:
added a comment in `CandidateForkConfig.Threshold` and a pointer here

Reply:
Couldn't we use a percentage value between 0.01 and 100.00 and parse it as a `BigDecimal`?